store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -492,6 +491,38 @@ public void modifyAttrs(Entry e, Map<String, ? extends Object> attrs,
         AttributeManager.getInstance().preModify(attrs, e, callbackContext, checkImmutable, allowCallback);
         modifyAttrsInternal(e, null, attrs);
         AttributeManager.getInstance().postModify(attrs, e, callbackContext, allowCallback);
+
+        postModificationValidation(e, attrs);
+
+    }
+
+    /**
+     * Performing post validation
+     * @param e
+     * @param attrs
+     * @throws ServiceException
+     */
+    private void postModificationValidation(Entry e, Map<String, ?> attrs) throws ServiceException {
+        if (e instanceof Account && !(e instanceof CalendarResource)) {
+            Account acct = (Account) e;
```

Comment:
Please reduce the redundant checks, type checking for `Account` and `CalendarResource` can be combined into one if statement which will improve readability, provide clearer separation of conditions, and offer more flexibility for future changes.

Reply:
Done